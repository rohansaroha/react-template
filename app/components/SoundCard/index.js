import React from 'react';
import PropTypes from 'prop-types';
import styled from 'styled-components';
import { Card, Input, Skeleton, Button } from 'antd';
import { Link } from 'react-router-dom';
import If from '@components/If';
import { colors, fonts } from '@app/themes';
import { injectIntl } from 'react-intl';
import { compose } from 'redux';

const CustomCard = styled(Card)`
  && {
    padding: 0;
    min-height: ${props => (props.complete ? '22em' : '16em')};
    position: relative;
    box-sizing: border-box;
    border-radius: 0.6em;
    margin: ${props => (props.complete ? '2em auto' : '0.6em auto')};
    flex-basis: ${props => (props.complete ? '45%' : '30%')};
  }
`;
const SearchBox = styled(Input)`
  && {
    margin: 0 auto;
    display: flex;
    justify-content: center;
  }
`;
const HeaderBox = styled.div`
  && {
    margin: 0 auto;
    display: flex;
    gap: 1em;
  }
`;
const SongPrimary = styled.div`
  && {
    max-height: 3em;
    overflow: hidden;
    margin: 0.6em 0 0.3em 0;
    ${props => (props.complete ? fonts.size.xRegular : fonts.size.regular)}
    color: ${colors.textPrimary};
  }
`;
const AudioImg = styled.img`
  && {
    max-height: ${props => (props.complete ? '12em' : '8em')};
    flex: ${props => (props.complete ? '2' : '1')};
    border-radius: 0.4em;
  }
`;
const SongSecondary = styled.div`
  && {
    overflow: hidden;
    margin-bottom: 1em;
    ${fonts.size.small};
    color: ${colors.textSecondary};
  }
`;
const AudioBox = styled.audio`
  && {
    width: 90%;
    position: absolute;
    bottom: 1em;
  }
`;
const ButtonSong = styled(Button)`
  && {
    margin: 2rem auto;
  }
`;
function SoundCard({ song, complete, loading, intl }) {
  const ButtonHandler = () => {
    window.location.href = song.trackViewUrl;
  };
  return (
    <CustomCard complete={complete} data-testid="sound-card">
      <Skeleton loading={loading} active>
        <div onClick={() => history.push(`/track/${song.trackId}`)} data-testid="link-track">
          <HeaderBox>
            <AudioImg src={song.artworkUrl100} alt={song.trackName} complete={complete} />
            <div style={{ flex: 4 }}>
              <SongPrimary complete={complete}>{song.trackName}</SongPrimary>
              <SongSecondary>{song.artistName}</SongSecondary>
              <If condition={complete}>
                <ButtonSong data-testid="play-button" danger onClick={ButtonHandler}>
                  {intl.formatMessage({ id: 'play_song' })}
                </ButtonSong>
              </If>
            </div>
          </HeaderBox>
        </div>
        <AudioBox controls>
          <source src={song.previewUrl} />
        </AudioBox>
      </Skeleton>
    </CustomCard>
  );
}
SoundCard.propTypes = {
  intl: PropTypes.object,
  song: PropTypes.object,
  complete: PropTypes.bool,
  loading: PropTypes.bool
};

export default compose(injectIntl)(SoundCard);
