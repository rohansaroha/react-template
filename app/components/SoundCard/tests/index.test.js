import React from 'react';
import { renderWithIntl } from '@utils/testUtils';
import SoundCard from '../index';

describe('<SoundCard /> tests', () => {
  let submitSpy;
  beforeEach(() => {
    submitSpy = jest.fn();
  });
  it('should render and match the snapshot', () => {
    const { baseElement } = renderWithIntl(<SoundCard />);
    expect(baseElement).toMatchSnapshot();
  });

  it('should contain 1 SoundCard component', () => {
    const { getAllByTestId } = renderWithIntl(<SoundCard />);
    expect(getAllByTestId('sound-card').length).toBe(1);
  });
});
