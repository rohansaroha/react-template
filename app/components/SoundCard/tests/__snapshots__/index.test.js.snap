// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`<SoundCard /> tests should ensure that complete data is being displayed 1`] = `
<body>
  <div>
    <div
      class="ant-card SoundCard__CustomCard-lvsa1p-0 guwSkq ant-card-bordered"
      data-testid="sound-card"
    >
      <div
        class="ant-card-body"
      >
        <div
          data-testid="link-track"
          style="cursor: pointer;"
        >
          <div
            class="SoundCard__HeaderBox-lvsa1p-2 elgeDp"
          >
            <img
              class="SoundCard__AudioImg-lvsa1p-4 jRZWqW"
            />
            <div
              style="flex: 4;"
            >
              <div
                class="SoundCard__SongPrimary-lvsa1p-3 djMiVS"
              />
              <div
                class="SoundCard__SongSecondary-lvsa1p-5 bBBuVf"
              />
              <button
                class="ant-btn SoundCard__ButtonSong-lvsa1p-7 gQYgvp ant-btn-dangerous"
                data-testid="play-button"
                type="button"
              >
                <span>
                  Play full Song
                </span>
              </button>
            </div>
          </div>
        </div>
        <audio
          class="SoundCard__AudioBox-lvsa1p-6 gwtEvh"
          controls=""
        >
          <source />
        </audio>
      </div>
    </div>
  </div>
</body>
`;

exports[`<SoundCard /> tests should match the snapshot when data is loaded  1`] = `
Array [
  <div
    data-testid="link-track"
    style="cursor: pointer;"
  >
    <div
      class="SoundCard__HeaderBox-lvsa1p-2 elgeDp"
    >
      <img
        class="SoundCard__AudioImg-lvsa1p-4 fGLDFf"
      />
      <div
        style="flex: 4;"
      >
        <div
          class="SoundCard__SongPrimary-lvsa1p-3 elLIt"
        />
        <div
          class="SoundCard__SongSecondary-lvsa1p-5 bBBuVf"
        />
      </div>
    </div>
  </div>,
]
`;

exports[`<SoundCard /> tests should match the snapshot when data is loading 1`] = `
<body>
  <div>
    <div
      class="ant-card SoundCard__CustomCard-lvsa1p-0 jruUYC ant-card-bordered"
      data-testid="sound-card"
    >
      <div
        class="ant-card-body"
      >
        <div
          class="ant-skeleton ant-skeleton-active"
        >
          <div
            class="ant-skeleton-content"
          >
            <h3
              class="ant-skeleton-title"
              style="width: 38%;"
            />
            <ul
              class="ant-skeleton-paragraph"
            >
              <li />
              <li />
              <li
                style="width: 61%;"
              />
            </ul>
          </div>
        </div>
      </div>
    </div>
  </div>
</body>
`;
